### PR TITLE
Re-plat on JsonExporter APIs and remove JsonSchemaMapper prototype

### LINKED ISSUE #56065: OpenAPI parameter descriptions are written to the wrong location
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

When a `[Description]` attribute is applied to a parameter of a Mimimal API endpoint intended to be documented with the OpenAPI functionality, the description string is written to the wrong place in the schema, which then means it is not documented by tools such as Swagger UI.

To illustrate, the JSON snippet below shows where the description should be (green) and where it currently appears (red):

```diff
{
  "name": "uppercase",
  "in": "query",
+ "description": "Whether to return the GUID in uppercase.",
  "schema": {
    "type": "boolean",
-   "description": "Whether to return the GUID in uppercase.",
    "nullable": true
  }
}
```

### Expected Behavior

The description is added to the top-level object for the parameter definition.

### Steps To Reproduce

Generate an OpenAPI document with `MapOpenApi()` for an endpoint similar to the below:

```csharp
app.MapGet("/name", Api.GetName);

public static class Api
{
    public static Ok<string> GetName([Description("Your name")] string name)
    {
        return TypedResults.Ok($"Hi {name}!");
    }
}

```

### Exceptions (if any)

_No response_

### .NET Version

9.0.100-preview.4.24267.66

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by captainsafia**
This pull request removes our copy of the `JsonSchemaMapper` prototype and replats the OpenAPI implementation onto the official `JsonSchemaExporter` APIs. There are a few deltas between the prototype implementation and the finalized implementation detailed here:

- The official implementation removes support for processing the `DescriptionAttribute`. We add this back and make sure that descriptions on parameters are set in the right location to close https://github.com/dotnet/aspnetcore/issues/56065.
- The official implementation removes support for processing nullability annotations on types. We add this back for both value and reference types.
- React to changes in API shape for the implementation: `JsonObject` -> `JsonNode`, `OnSchemaGenerated` -> `TransformSchemaNode `.
- Remove `pattern` property set by underlying schema generation for primitive types like `DateTime` and `integer` to match existing conventions in OpenAPI docs.

### CODE REVIEW

File: src/OpenApi/test/Integration/snapshots/OpenApiDocumentIntegrationTests.VerifyOpenApiDocument_documentName=responses.verified.txt [hunk 1]
```diff
@@ -78,30 +78,7 @@
             "content": {
               "application/json": {
                 "schema": {
-                  "type": "object",
```
Comment (captainsafia) [inline src/OpenApi/test/Integration/snapshots/OpenApiDocumentIntegrationTests.VerifyOpenApiDocument_documentName=responses.verified.txt:81]: This change is a result of new behavior in the underlying schema generator that removes the `anyOf` property for polymorphic types that don't contain any visible properties. In this case, all the properties on `Shape` are internal given that our implementation doesn't fully support polymorphic types yet which is why this schema is currently empty. This work was pending the incoming $ref version.

**Review by BrennanConroy — APPROVED** [review]

File: src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs [hunk 2]
```diff
@@ -22,8 +22,10 @@ namespace Microsoft.AspNetCore.OpenApi;
 /// Provides a set of extension methods for modifying the opaque JSON Schema type
 /// that is provided by the underlying schema generator in System.Text.Json.
 /// </summary>
-internal static class JsonObjectSchemaExtensions
+internal static class JsonNodeSchemaExtensions
 {
+    private static readonly NullabilityInfoContext _nullabilityInfoContext = new();
```
Comment (eiriktsarpalis) [inline src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:27]: `NullabilityInfoContext` isn't thread safe and hasn't been designed to be used as a singleton. I would recommend allocating a fresh instance for every type graph you are looking to generate.

File: src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs [hunk 3]
```diff
@@ -329,10 +346,48 @@ internal static void ApplyPolymorphismOptions(this JsonObject schema, JsonSchema
...
+    /// <param name="schema">The <see cref="JsonNode"/> produced by the underlying schema generator.</param>
+    /// <param name="parameterInfo">The <see cref="ParameterInfo" /> associated with the schema.</param>
+    internal static void ApplyNullabilityContextInfo(this JsonNode schema, ParameterInfo parameterInfo)
+    {
+        if (parameterInfo.ParameterType.IsValueType)
+        {
+            return;
+        }
+
+        var nullabilityInfo = _nullabilityInfoContext.Create(parameterInfo);
```
Comment (eiriktsarpalis) [inline src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:368]: Nullability information is stored in the `JsonPropertyInfo`/`JsonParameterInfo` types, so you can avoid taking a dependency on a reflection-based components that might be producing inconsistent results in trimmed applications.

Comment (captainsafia) [reply src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:368]: > Nullability information is stored in the `JsonPropertyInfo`/`JsonParameterInfo` types, so you can avoid taking a dependency on a reflection-based component that might be producing inconsistent results in trimmed applications.

Are you referring to `isGetNullable`/`IsSetNullable` here?

Comment (eiriktsarpalis) [reply src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:368]: Correct

Comment (captainsafia) [reply src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:368]: Got it!

How do we resolve JsonParameterInfo? We're no longer passing ParameterInfo to the exporter API. Is there an explicit API for resolving a JsonParameterInfo instance?

Comment (eiriktsarpalis) [reply src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:368]: Yep, it's now a property on JsonPropertyInfo itself.

Comment (captainsafia) [reply src/OpenApi/src/Extensions/JsonNodeSchemaExtensions.cs:368]: > Yep, it's now a property on JsonPropertyInfo itself.

Ah, I dug into this. It looks like this applies specifically for constructor parameters. For our particular case, we care about parameters on a method.

**Pull request merged**